tests/Feature/MyParcelComApiTest.php

```diff
@@ -112,6 +112,60 @@ public function testCreateMinimumViableShipment()
         );
     }
 
+    /** @test */
+    public function testSaveShipment()
+    {
+        $recipient = (new Address())
+            ->setFirstName('Bobby')
+            ->setLastName('Tables')
+            ->setCity('Birmingham')
+            ->setStreet1('Newbourne Hill')
+            ->setStreetNumber(12)
+            ->setPostalCode('B48 7QN')
+            ->setCountryCode('GB');
+
+        // Minimum required data should be a recipient address and weight. All
+        // other data should be filled with defaults.
+        $shipment = (new Shipment())
+            ->setWeight(500)
+            ->setRecipientAddress($recipient);
+
+        $shipment = $this->api->saveShipment($shipment);
+
+        $this->assertNotNull(
+            $shipment->getServiceContract(),
+            'When no service contract has been selected, the preferred service contract for given shipment should be used'
+        );
+        $this->assertNotNull(
+            $shipment->getId(),
+            'Once the shipment has been created, it should have an id'
+        );
+        $this->assertNotNull(
+            $shipment->getPrice(),
+            'Successfully created shipments should have a price'
+        );
+        $this->assertEquals(
+            $this->api->getDefaultShop()->getReturnAddress(),
+            $shipment->getSenderAddress(),
+            'The shipment\'s sender address should default to the default shop\'s return address'
+        );
+        $this->assertEquals(
+            $recipient,
+            $shipment->getRecipientAddress(),
+            'The shipment\'s recipient address should not have changed'
+        );
+
+        // Save an existing shipment should patch it
+        $shipment = $this->api->saveShipment($shipment);
+
+        $this->assertEquals(
+            'new patched description',
+            $shipment->getDescription(),
+            'patch dit not return a shipment with a new description'
+        );
```

Comment:
This last test seems a bit off to me. Now it seems like that whenever you save an existing shipment, it's description should automatically get updated. This however is not the case.
Instead the mocked client should simulate what the patch endpoint would do, and that would be to merge the request with the original data and return that. You can do this by checking in the `MocksApiCommunication` trait if the request is a `patch` request and merge the request with the original data (which would be in the json stub, maybe even just have it be a simlink to the one in the get directory).